app/assets/javascripts/tramway/multiselect_controller.js

```diff
@@ -0,0 +1,136 @@
+    selectedItemTemplate: String,
+    dropdownState: String,
+    selectedItems: Array,
+    placeholder: String,
+    selectAsInput: String,
+    value: Array
+  }
+
+  connect() {
+    this.dropdownState = 'closed';
+    this.unselectedItems = JSON.parse(this.element.dataset.items);
+
+    const initialValues = this.element.dataset.value === undefined ? [] : this.element.dataset.value.split(',')
+    this.selectedItems = this.unselectedItems.filter(item => initialValues.includes(item.value.toString()));
+    this.unselectedItems = this.unselectedItems.filter(item => !initialValues.includes(item.value.toString()));
+
+    this.renderSelectedItems();
+  }
+
+  renderSelectedItems() {
+    const allItems = this.fillTemplate(this.element.dataset.selectedItemTemplate, this.selectedItems);
+    this.showSelectedAreaTarget.innerHTML = allItems;
+    this.showSelectedAreaTarget.insertAdjacentHTML("beforeEnd", this.input());
+    this.updateInputOptions();
+  }
+
+  fillTemplate(template, items) {
+    return items.map((item) => {
+      return template.replace(/{{text}}/g, item.text).replace(/{{value}}/g, item.value)
+    }).join('')
+  }
+
+  closeOnClickOutside(event) {
+    if (this.dropdownState === 'open' && !this.element.contains(event.target)) {
+      this.closeDropdown();
+    }
+  }
+
+  toggleDropdown() {
+    if (this.dropdownState === 'closed') {
+      this.openDropdown();
+    } else {
+      this.closeDropdown();
+    }
+  }
+
+  rerenderItems() {
+    this.closeDropdown();
+    this.openDropdown();
+  }
+
+  openDropdown() {
+    this.dropdownState = 'open';
+    this.dropdownTarget.insertAdjacentHTML("afterend", this.template);
+
+    if (this.dropdown()) {
+      this.dropdown().addEventListener('click', event => event.stopPropagation());
+    }
+  }
+
+  dropdown() {
+    return this.element.querySelector('#dropdown');
+  }
+
+  closeDropdown() {
+    this.dropdownState = 'closed';
+    if (this.dropdown()) {
+      this.dropdown().remove();
+    }
+  }
+
+  get template() {
+    return this.element.dataset.dropdownContainer.replace(
+      /{{content}}/g,
+      this.fillTemplate(this.element.dataset.itemContainer, this.unselectedItems)
+    );
+  }
+
+  toggleItem({ currentTarget }) {
+    const item = {
+      text: currentTarget.dataset.text,
+      value: currentTarget.dataset.value
+    };
+
+    const itemIndex = this.selectedItems.findIndex(x => x.value.toString() === item.value);
+    if (itemIndex !== -1) {
+      this.selectedItems = [...this.selectedItems.slice(0, itemIndex), ...this.selectedItems.slice(itemIndex + 1)];
+    } else {
+      this.selectedItems = [...this.selectedItems, item];
+    }
+
+    const unselectedItemIndex = this.unselectedItems.findIndex(x => x.value.toString() === item.value);
+    if (unselectedItemIndex !== -1) {
+      this.unselectedItems = [
+        ...this.unselectedItems.slice(0, unselectedItemIndex),
+        ...this.unselectedItems.slice(unselectedItemIndex + 1)
+      ];
+    }
```

Comment:
it looks like just filter to me
`this.unselectedItems = this.unselectedItems.filter(x => x.value !== value);`

Reply:
fixed :slightly_smiling_face: